hoodie-common/src/main/java/com/uber/hoodie/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -113,13 +123,12 @@ public HoodieLogBlockType getBlockType() {
     return HoodieLogBlockType.AVRO_DATA_BLOCK;
   }
 
-  public static HoodieLogBlock fromBytes(byte[] content, Schema readerSchema, boolean readMetadata)
-      throws IOException {
+  public static HoodieLogBlock fromBytes(byte[] content, Schema readerSchema, boolean readMetadata) throws IOException {
 
-    DataInputStream dis = new DataInputStream(new ByteArrayInputStream(content));
+    SizeAwareDataInputStream dis = new SizeAwareDataInputStream(new DataInputStream(new ByteArrayInputStream(content)));
```

Comment:
This usage led me to expect SizeAwareDIS be a subclass of DataInputStream.. Should it be?

Reply:
We cannot make it a subclass since methods in DataInputStream are final.

Reply:
ah. got it..